### DISCUSSION TITLE
[ideas] geotargets package for geospatial formats and example workflows

### DISCUSSION THREAD

**Discussion by njtierney** [ropensci/targets#1248] [Ideas]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Basically the idea is that we create a target package extension that works for geospatial formats.

Some of the ideas and problems/solutions have been written about here: https://github.com/njtierney/demo-geospatial-targets/

The proposal of an R package was formalised in https://github.com/njtierney/demo-geospatial-targets/issues/3 by https://github.com/aariq

**Comment by wlandau** [comment]
Great idea! In case you need target factories like in `tarchetypes`, here is what I recommend: https://wlandau.github.io/targetopia/contributing.html

**Comment by Aariq** [comment]
Woo hoo! I'm happy to help as I'll likely get a lot of benefit from this!  I think `geotargets` is a great name as well.  Please tag me when a repo exists and we can discuss some ideas

**Comment by Aariq** [comment]
This would be a good candidate for an [R Consortium ISC grant](https://www.r-consortium.org/all-projects/call-for-proposals) as well.  Next grant application period closes April 1, 2024.  Not sure if you're in a position to move that quickly or are interested.  I've done two of these now—one with another co-maintainer—and I'd be happy to chat about the process.

**Comment by njtierney** [comment]
Awesome, I'm keen to write a R Consortium grant - while I'm an experienced R package creator/developer, I'm not that well versed on the geospatial parts of this. I'd be happy to get it started though, and happy to co-write.

**Reply by Aariq** [reply]
hmm... I'm not super well versed in the geospatial bits either.  Would be good to find someone geospatial but maybe newer to `targets` to help out

**Reply by Aariq** [reply]
Although part of the grant funding could be for gathering info from the r-spatial community about needs and commonly used formats for intermediate steps and such.  With the proposal we wrote for `webchem`, we included an information gathering phase like this which was basically social media polls and emails sent out to listservs.

**Comment by njtierney** [comment]
Here's the initial repo: https://github.com/njtierney/geotargets! Still needing to work on larger implementation information.

**Reply by wlandau** [reply]
Exciting stuff!
